internal/contributebot/main.go

```diff
@@ -131,6 +133,33 @@ func (w *worker) receiveIssueEvent(ctx context.Context, e *github.IssuesEvent) e
 	return edits.Execute(ctx, client, owner, repo, issueNumber)
 }
 
+func (w *worker) receivePullRequestEvent(ctx context.Context, e *github.PullRequestEvent) error {
+
+	// Pull out the interesting data from the event.
+	data := &pullRequestData{
+		Action:      e.GetAction(),
+		Repo:        e.GetRepo().GetName(),
+		PullRequest: e.GetPullRequest(),
+		Change:      e.GetChanges(),
+	}
+
+	// Refetch the pull request in case the event data is stale.
+	client := w.ghClient(e.GetInstallation().GetID())
+	owner := e.GetRepo().GetOwner().GetLogin()
+	repo := e.GetRepo().GetName()
+	prNumber := data.PullRequest.GetNumber()
+	pr, _, err := client.PullRequests.Get(ctx, owner, repo, prNumber)
+	if err != nil {
+		return err
+	}
+	data.PullRequest = pr
+
+	// Process the pull request, deciding what actions to take (if any).
+	edits := processPullRequestEvent(data)
+	// Execute the actions (if any).
+	return edits.Execute(ctx, client, owner, repo, prNumber)
```

Comment:
It seems to me that some arguments can be passed by a *PullRequestEvent and extract inside Execute, would that be clearer?

Reply:
Now passing data to Execute as well, and xxxData struct has everything that Execute needs.